spec/solver_spec.rb

```diff
@@ -0,0 +1,22 @@
+require_relative '../solver'
+
+describe Solver do
+  it 'returns factorial of a number and tracks constraints' do
+    solver = Solver.new
+    expect(solver.factorial(5)).to eq(120)
+    expect(solver.factorial(0)).to eq(1)
+  end

```

Comment:
- [ ] It would be better if our factorial method tests for a negative number as in the code review guidelines below
![Screenshot from 2022-07-28 13-36-15](https://user-images.githubusercontent.com/46388113/181518875-64da380f-6d92-4644-9855-3dd7f9ee7a10.png)